src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -764,4 +765,17 @@ default int getDefaultTooltipHideFlags(@Nonnull ItemStack stack)
         return 0;
     }
 
+    /**
+     * Get the food properties for this item.
+     * Use this instead of the {@link Item#getFoodProperties()} method, for ItemStack sensitivity.
+     *
+     * @param stack The ItemStack the entity wants to eat.
+     * @param entity The entity which wants to eat the food. Be aware that this can be null!
+     * @return The current FoodProperties for the item.
+     */
+    default FoodProperties getFoodProperties(ItemStack stack, @Nullable LivingEntity entity)
+    {
+        return self().getFoodProperties();
```

Comment:
The base food properties method should be deprecated with a comment/javadoc saying to use this more sensitive version.